File: GameLayers/pickAttackCardLayer.js
import {config, print, debug} from '../config.js'

export class OneCardPickAttackCardsLayer{
    constructor(policy){
        this.policy = policy
    }

    execute(world){
        world.attackCards = [this.policy.pickCards(world.heroHand, world)[0]]
        print('Picked Cards:', world.attackCards.map(n => n.attack))
    }
}

export class MultipleCardPickAttackCardsLayer{
    constructor(policy){
        this.policy = policy
    }

    execute(world){
        world.attackCards = this.policy.pickCards(world.heroHand, world)
        print('Picked Cards:', world.attackCards.map(n => n.attack))
    }
}

File: python/GameLayers/pickAttackCardLayer.py
import config

class OneCardPickAttackCardsLayer:
    def __init__(self, policy):
        self.policy = policy

    def execute(self, world):
        world.attackCards = [self.policy.pickCards(world.heroHand, world)[0]]
        # log('Picked Cards:', world.attackCards.map(n => n.attack))

class MultipleCardPickAttackCardsLayer:
    def __init__(self, policy):
        self.policy = policy

    def execute(self, world):
        world.attackCards = self.policy.pickCards(world.heroHand, world)
        # log('Picked Cards:', world.attackCards.map(n => n.attack))
        



